03-events-forms-refs-life-cycle/forms-and-refs/part-1-divs/src/App.js

```diff
@@ -0,0 +1,209 @@
+import React, { Component } from 'react';
+import './App.css';
+import CustomDiv from './CustomDiv';
+import DivForm from './DivForm';
+
+class App extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      divs : [{height: 500, width: 400, color : 'red', key : 0}],
+      height : '',
+      width : '',
+      color : '',
+      key : 0 
+    }
+  }
+
+  handleChange(e){
+    e.preventDefault();
+    this.setState({[e.target.name] : e.target.value})
+  }
+
+  handleSubmit(){
+    let height = Math.min(Math.max(this.state.height, 50), 500);
+    let width = Math.min(Math.max(this.state.height, 50), 500);
+    let color = this.props.allColors.includes(this.state.color) ? this.state.color : 'grey';
+    let key = this.state.key + 1;
+    this.setState({ key });
+    this.setState({divs : [...this.state.divs,{height, width, color, key}]});
+
+  }
+
+
+  render() {
+    let divs = this.state.divs.map(div => {
+      return <CustomDiv key={div.key} height={div.height} width={div.width} color={div.color} />
+    })
+
+
+    return (
+      <div className="App">
+        <DivForm 
+          height={this.state.height} 
+          width={this.state.width} 
+          color={this.state.color} 
+          handleChange={this.handleChange.bind(this)} 
```

Comment:
these could be bound in the constructor since you have no argument binding, just context binding